modules/post/windows/gather/checkvm.rb

```diff
@@ -170,24 +183,56 @@ def xen?
       return true if service_exists?(service)
     end
 
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemProductName') =~ /xen/i
+
     false
   end
 
   def qemu?
-    key_path = 'HKLM\HARDWARE\DEVICEMAP\Scsi\Scsi Port 0\Scsi Bus 0\Target Id 0\Logical Unit Id 0'
+    key_path = 'HKLM\\HARDWARE\\DEVICEMAP\\Scsi\\Scsi Port 0\\Scsi Bus 0\\Target Id 0\\Logical Unit Id 0'
     return true if registry_getvaldata(key_path, 'Identifier') =~ /qemu|virtio/i
 
-    key_path = 'HKLM\HARDWARE\DESCRIPTION\System\CentralProcessor\0'
+    key_path = 'HKLM\\HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0'
     return true if registry_getvaldata(key_path, 'ProcessorNameString') =~ /qemu/i
 
-    %w[HKLM\HARDWARE\ACPI\DSDT HKLM\HARDWARE\ACPI\FADT HKLM\HARDWARE\ACPI\RSDT].each do |key|
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'VideoBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemManufacturer') =~ /qemu/i
+
+    %w[HKLM\\HARDWARE\\ACPI\\DSDT HKLM\\HARDWARE\\ACPI\\FADT HKLM\\HARDWARE\\ACPI\\RSDT].each do |key|
       srvvals = registry_enumkeys(key)
       return true if srvvals && srvvals.include?('BOCHS_')
     end
 
     false
   end
 
+   def parallels?
```

Comment:
Hey @jvoisin, thanks for enhancements 🙏 

Did some testing on Windows 11 VM running in Parallels. Found a couple issues:

1. The first time I ran the windows `checkvm` module the hypervisor was reported as `Hyper-V`. The service `vmicheartbeat` is running in the Parallels vm and satisfies the check on line `81`:
```
[7] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>):1> %w[vmicexchange vmicheartbeat vmicshutdown vmicvss].each do |service|
[7] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>):1*   return true if service_exists?(service)
[7] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>):1* end
[+] This is a Hyper-V Virtual Machine
[*] Post module execution completed
```


2. I rearranged the order of the `if` statements so `if parallels?` would be run before `if hyperv?`. The hypervisor was then reported as `VirtualBox`  ( `parallels?` returned false ). The check currently on line `160` is too loose:
```
[2] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemProductName')
=> "Parallels Virtual Platform"
[3] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemProductName') =~ /virtual/i
[+] This is a VirtualBox Virtual Machine
[*] Post module execution completed
```

3. The `registry_getvaldata` returns an array when querying `SystemBiosVersion` in Parallels:
```
[2] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion')
=> ["PRLS   - 1", "18.3.2 (53621)", "Parallels EFI x64 18.3.2-53621 - 12D175"]
```

Adding `.join` to the query solves the issue:
```
[2] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion').join =~ /parallels/i
[+] This is a Parallels Virtual Machine
[*] Post module execution completed
```

However I'm not certain this will always return an array. It might be best to check if it's an array, and if so, use `.joins`.

The `Data` field of `SystemBiosVersion` appears to be a single string, however when you go to edit it you can see there are new lines in the string which split it into multiple array elements. Wondering if this issue might see this same issue with some of the other hypervisors.  

![Screenshot 2023-07-17 at 12 31 03 PM](https://github.com/rapid7/metasploit-framework/assets/23320005/d0f6d57a-f6af-4f41-ae7b-c4002f46deea)